include/SoapySDR/Types.hpp

```diff
@@ -173,16 +174,19 @@ namespace Detail {
 template <typename Type>
 typename std::enable_if<std::is_same<Type, bool>::value, Type>::type StringToSetting(const std::string &s)
 {
-    if (s == SOAPY_SDR_TRUE) return true;
-    if (s == SOAPY_SDR_FALSE) return false;
-
-    //zeros and empty strings are false
-    if (s == "0") return false;
-    if (s == "0.0") return false;
-    if (s == "") return false;
-
-    //other values are true
+  if (s.empty() || s == SOAPY_SDR_FALSE) {
+    return false;
+  }
+  if (s == SOAPY_SDR_TRUE) {
```

Comment:
*Cough* *cough* `|| -> or`

We do `#include <ciso646>` here :-)

Reply:
> `|| -> or`

oh, if that's the style of this project! (not what I'd usually do.)

> ciso646

ah, that is part of C++ anyway, IIRC, and was only necessary to get `or` in C, right? (or does MSVC bite us again?)

Reply:
no, wait, even microsoft says it's deprecated and might be removed (that I believe when I see it): https://docs.microsoft.com/en-us/cpp/standard-library/ciso646?view=msvc-160

Reply:
No worries, whatever makes the compiler happy will get ifdef'd here:
https://github.com/pothosware/SoapySDR/blob/master/include/SoapySDR/Config.hpp#L13

I think of the countless gnuradio patches to switch boolean operators back to `&&` or `||` and I wish someone just quietly snuck this include somewhere into a main header files. ;-)